server/src/main/java/io/seata/server/session/GlobalSession.java

```diff
@@ -246,10 +246,12 @@ public void addBranch(BranchSession branchSession) throws TransactionException {
 
     @Override
     public void removeBranch(BranchSession branchSession) throws TransactionException {
+        if (!branchSession.unlock()) {
+            throw new RuntimeException("Unlock branch lock failed!");
```

Comment:
throw TransactionException, not RuntimeException, otherwise, some forEach will break.

Reply:
done